library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
What does prevent cache flushing after the sort to actually make cache empty?
The first writer checks whether cache is empty and doesn't start any binlog, while another writer in parallel inserts new values into the cache.

Frankly, this whole patch looks like overoptimizing a feature which is not a bottleneck at all. And while it could safe, there are so many questions, that I would rather have a really tiny startup initialization delay than corrupting data, which might be caused by the race between cached value and on-disk changes.

Let's not introduce this headache, I agree - additional index iteration takes few seconds under lock, so I will update this chunk just to skip that part if cache is empty, it is indeed useless and is performed under lock.